### PR TITLE
Add getting started documents for IDP

### PR DESCRIPTION
**Pull request by OrkoHunter**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

Add more docs prior to Beta launch.

Preview URL - https://6480711c45bbf00c48e37f16--harness-developer.netlify.app/

### CODE REVIEW

File: docs/internal-developer-portal/getting-started/enable-a-new-plugin.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+1. All the admin configurations are provided in the Admin panel. Once users with admin permissions are onboarded, they can access the Admin option in the side nav. Clicking on the Admin section will navigate the user to the Admin view, where they can see multiple options such as Plugins, Layout, Connectors, Configurations, Access Control, and more.
+
+![](static/plugin-page-nav.png)
+
+2. The "Plugins" navigation option will direct the user to the plugins list page, which showcases a collection of curated plugins. On this page, there are distinct sections dedicated to both enabled and disabled plugins, providing a clear differentiation between the two categories.
+
+![](static/att_4_for_21398290667.png)
+
+3. In the Plugins navigation, click on the desired plugin card. This will redirect users to the Plugin details screen, which provides information about the plugin such as its name, details, plugin layout, configurations, and the option to add a plugin secret.
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/getting-started/enable-a-new-plugin.md:49]:
```suggestion
3. Select the desired plugin card. 
The Plugin details page is displayed. The page provides information such as the plugin's name, details, layout, and configurations. It also includes an option to add a plugin secret. The plugin's layout section is read-only and displays the exported cards, tabs, or pages provided by the plugin.
```

File: docs/internal-developer-portal/getting-started/enable-a-new-plugin.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+![](static/att_5_for_21398290667.png)
+
+![](static/att_3_for_21398290667.png)
+
+4. The plugin's layout section is read-only and displays the exported cards, tabs, or pages provided by the plugin.
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/getting-started/enable-a-new-plugin.md:55]: Not a step. Merged into the previous step's info section. Remove and renumber.

File: docs/internal-developer-portal/getting-started/enable-a-new-plugin.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+5. In the configuration section, a default config YAML is provided, which can be edited using the YAML editor. Users have the option to customize the configuration based on their requirements.
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/getting-started/enable-a-new-plugin.md:57]:
```suggestion
5. (Optional) If you want to customize the plugin's configuration, in the **Configurations** section, edit the default configuration YAML. 
```

File: docs/internal-developer-portal/getting-started/enable-a-new-plugin.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+6. Below the configuration, users can update the plugin secret token value as specified in the config YAML. Make sure the secret token name is the same as given in the config YAML.
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/getting-started/enable-a-new-plugin.md:59]:
```suggestion
6. Update the plugin secret token value as specified in the configuration YAML. Make sure that the secret token name is the same as given in the configuration YAML.
```

File: docs/internal-developer-portal/getting-started/enable-a-new-plugin.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+7. After verifying all the configurations and secret details, users can save the configuration. If the config YAML is valid, the plugin configuration will be saved successfully. Otherwise, appropriate error messages will be displayed.
+
+8. Upon saving the configuration, the "Enable Plugin" button will automatically be enabled. Users can enable the plugin by clicking on this button.
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/getting-started/enable-a-new-plugin.md:63]:
```suggestion
8. To enable the plugin, select **Enable Plugin**. 
The plugin is moved to the Enabled Plugins section on the Plugins page.
```

File: docs/internal-developer-portal/getting-started/enable-a-new-plugin.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+9. Once the selected plugin is enabled, it will be moved to the "Enabled Plugins" section on the Plugins List page. Users can navigate to this page by clicking on the "Back" button.
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/getting-started/enable-a-new-plugin.md:65]: Remove the first sentence. I added this sentence to the additional info section of the previous step. 
```suggestion
9. To return to the Plugins page, select **Back**.
```

File: docs/internal-developer-portal/getting-started/enable-a-new-plugin.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+10. After enabling the plugin, users are required to ingest the layout using plugin layout management. The layout should be modified according to the exported elements from the plugin, which may include cards, tabs, and pages.
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/getting-started/enable-a-new-plugin.md:67]:
```suggestion
10. Ingest the layout by using plugin layout management. The layout should be modified according to the elements that are exported from the plugin. The elements might include cards, tabs, and pages.
```

File: docs/internal-developer-portal/getting-started/enable-a-new-plugin.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+For instance, if the Jenkins plugin exports one card and one tab, in order to display them in your catalog entity, the user must update the layout accordingly. To add a dedicated tab for the Jenkins plugin, follow these steps:
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/getting-started/enable-a-new-plugin.md:69]:
```suggestion
For instance, if the Jenkins plugin exports one card and one tab, in order to display them in your catalog entity, you must update the layout accordingly. To add a dedicated tab for the Jenkins plugin, follow these steps:
```

File: docs/internal-developer-portal/getting-started/enable-a-new-plugin.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+- From the dropdown menu, select the domain option.
+- Search for the "ci-cd" component within the domain.
+- Inside the Entity Switch case, insert the following code snippet.
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/getting-started/enable-a-new-plugin.md:73]:
```suggestion
- Inside the **Entity Switch** case, insert the following code snippet:
```

File: docs/internal-developer-portal/getting-started/enable-a-new-plugin.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+![](static/layout-snippet.png)
+
+These are the simple steps to enable a plugin in Harness IDP. Finally, the user will be able to view the enabled plugins in the catalog section of Harness IDP. To see the enabled plugins, the user simply needs to register the plugin catalog info YAML.
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/getting-started/enable-a-new-plugin.md:77]:
```suggestion
You should now be able to view the enabled plugins in the catalog section of Harness IDP. To see the enabled plugins, register the plugin catalog info YAML.
```

File: docs/internal-developer-portal/getting-started/enable-docs.md [hunk 11]
```diff
@@ -0,0 +1,80 @@
+---
+title: Enable the Docs for your component
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/getting-started/enable-docs.md:2]:
```suggestion
title: Enable documentation for your component
```

File: docs/internal-developer-portal/getting-started/enable-docs.md [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+description: Learn how you can enable the TechDocs plugin to show your markdown docs under the "Docs" tab on the catalog page
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/getting-started/enable-docs.md:3]:
```suggestion
description: Learn how you can enable the TechDocs plugin to show your markdown docs on the Docs tab of the catalog page.
```

File: docs/internal-developer-portal/getting-started/enable-docs.md [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+sidebar_position: 40
+helpdocs_topic_id:
+helpdocs_category_id:
+helpdocs_is_private: false
+helpdocs_is_published: true
+---
+
+By default, the "Docs" tab in your catalog will be empty for a new software component. However, let's quickly get our markdown docs rendered for the component on the "Docs" page in the catalog.
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/getting-started/enable-docs.md:11]:
```suggestion
By default, the Docs tab in your catalog does not include documentation for a new software component. However, you can quickly publish Markdown documentation to the Docs tab.
```
Global comment: do not enclose names of UI elements (in this case, the Docs tab) in quotation marks.

File: docs/internal-developer-portal/getting-started/enable-docs.md [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+
+![](static/docs-empty.png)
+
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/getting-started/enable-docs.md:14]: To publish documentation to the Docs tab, follow these steps:

File: docs/internal-developer-portal/getting-started/register-a-new-software-component.md [hunk 15]
```diff
@@ -0,0 +1,63 @@
+---
+title: Add a new Software Component in the catalog
+description: Learn how you can add a new software component in the IDP Software Catalog.
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/getting-started/register-a-new-software-component.md:3]:
```suggestion
description: Learn how you can add a new software component in the IDP software catalog.
```

File: docs/internal-developer-portal/getting-started/register-a-new-software-component.md [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+      type: admin-dashboard
+spec:
+  type: service
+  lifecycle: production
+  owner: team-a
+  system: project-x
+```
+
+The key fields for you to update here are
+
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/getting-started/register-a-new-software-component.md:40]: Replace numbering with bullets. This list should be an unordered list.

File: docs/internal-developer-portal/getting-started/register-a-new-software-component.md [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+1. `metadata.name` - This should be a unique name of your component, usually the name of the service.
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/getting-started/register-a-new-software-component.md:41]:
```suggestion
1. `metadata.name`. This should be a unique name for your component. Usually, it is the name of the service.
```
(Also replace hyphens with periods or colons. I've selected the period here and you can too.)

File: docs/internal-developer-portal/getting-started/register-a-new-software-component.md [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+2. `metadata.description` - The description of your new component.
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/getting-started/register-a-new-software-component.md:42]:
```suggestion
2. `metadata.description` - A description for your new component.
```

File: docs/internal-developer-portal/getting-started/register-a-new-software-component.md [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+3. `spec.type` - The new software component could be a `service`, `library`, `website` or something else.
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/getting-started/register-a-new-software-component.md:43]:
```suggestion
3. `spec.type` - The new software component could be a `service`, `library`, `website`, or any other type.
```

File: docs/internal-developer-portal/getting-started/register-a-new-software-component.md [hunk 20]
```diff
@@ -0,0 +1,63 @@
...
+4. `spec.owner` - The user group identifier of the team who owns the component.
+
+Once the file is created on your git, copy the full URL to the file e.g. `https://github.com/harness-community/idp-samples/blob/main/catalog-info.yaml`.
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/getting-started/register-a-new-software-component.md:46]:
```suggestion
Once the file is created in your Git repo, copy the full URL to the file. For example, `https://github.com/harness-community/idp-samples/blob/main/catalog-info.yaml`.
```

**Review by deepakbolangady — APPROVED** [review]
Given the time crunch, I'm approving with comments, but it's likely that I'll have to approve again. 
I have not had a chance to look at the topic for setting up IDP, and I'll need another stab at the topic for registering a software component (it needs a reorg with numbering). I'll take a look at these issues tomorrow morning. If there is time to incorporate, please do. Otherwise, we can come back to it as soon as we can.

File: docs/internal-developer-portal/getting-started/enable-a-new-plugin.md [hunk 21]
```diff
@@ -0,0 +1,77 @@
...
+title: Enable a plugin in Harness IDP
+description: Learn how you to enable a new plugin in Harness IDP and how to make use of it in your Software Catalog
+sidebar_position: 50
+helpdocs_topic_id:
+helpdocs_category_id:
+helpdocs_is_private: false
+helpdocs_is_published: true
+---
+
+Enabling plugins in Harness IDP allows you to extend its functionality and customize it to suit your specific needs. This guide will provide step-by-step instructions on how to enable a new plugin in Harness IDP.
```
[the remark this answers is not in the document]
Comment (OrkoHunter) [reply docs/internal-developer-portal/getting-started/enable-a-new-plugin.md:11]: Yes - plugins extend IDP's functionality. Thanks!

File: docs/internal-developer-portal/getting-started/enable-a-new-plugin.md [hunk 22]
```diff
@@ -0,0 +1,77 @@
...
+
+**Process:**
+
+To enable a plugin, users can follow these steps:
+
+1. All the admin configurations are provided in the Admin panel. Once users with admin permissions are onboarded, they can access the Admin option in the side nav. Clicking on the Admin section will navigate the user to the Admin view, where they can see multiple options such as Plugins, Layout, Connectors, Configurations, Access Control, and more.
+
+![](static/plugin-page-nav.png)
+
+2. The "Plugins" navigation option will direct the user to the plugins list page, which showcases a collection of curated plugins. On this page, there are distinct sections dedicated to both enabled and disabled plugins, providing a clear differentiation between the two categories.
```
[the remark this answers is not in the document]
Comment (OrkoHunter) [reply docs/internal-developer-portal/getting-started/enable-a-new-plugin.md:45]: Got it, I'll adjust the numberings.

File: docs/internal-developer-portal/getting-started/enable-docs.md (hunk 13 above)
Comment (OrkoHunter) [reply docs/internal-developer-portal/getting-started/enable-docs.md:11]: Got it, thanks!

**Comment by KimberlyFields** [comment]
@OrkoHunter I'm going to do some additional review to help Deepak. I'll go ahead and commit small changes to grammar, punctuation, and style. I'll comment where I have questions.

**Review by deepakbolangady — APPROVED** [review]
Approving for a timely merge before PM needs it today. 
Pending tasks:
* Setting up IDP needs a review
* Another stab at the topic for registering a software component (it needs a reorg with numbering).

**Review by deepakbolangady — APPROVED** [review]
Approving so that we can merge once before the demo/onboarding.

Pending:
* docs/internal-developer-portal/getting-started/setting-up-idp.md needs an overhaul.
* docs/internal-developer-portal/getting-started/register-a-new-software-component.md needs a reorg with numbering.

**Pull request merged**